tb-profiler/tb-profiler-profile.cwl

```diff
@@ -0,0 +1,227 @@
+      inputBinding:
+        prefix: --platform
+        position: 40      
+    default: "illumina"
+  output_prefix:
+    label: "Prefix to use for all results generated"
+    type: string?
+    default: "tbprofiler"
+    inputBinding:
+      prefix: --prefix
+      position: 50
+  csv_output:
+    label: "Produce CSV output"
+    type: boolean?
+    inputBinding:
+      prefix: --csv
+      position: 50
+  text_output:
+    label: "Produce text output"
+    type: boolean?
+    inputBinding:
+      prefix: --txt
+      position: 50
+  pdf_output:
+    label: "Produce PDF output (using pdflatex)"
+    type: boolean?
+    inputBinding:
+      prefix: --pdf
+      position: 50
+  mapper:
+    label: "Mapping tool to use."
+    doc: "Defaults to bwa if the illumina sequencing platform is selected, and defaults to minimap2 for nanopore"
+    type:
+      - type: enum
+        symbols:
+          - "bwa"
+          - "minimap2"
+          - "bowtie2"
+          - "bwa-mem2"
+        inputBinding:
+          prefix: --mapper
+          position: 60
+      - "null"      
+  caller:
+    label: "Variant calling tool to use"
+    doc: "Default is bcftools"
+    type:
+      - type: enum
+        symbols:
+          - "bcftools"
+          - "gatk"
+          - "freebayes"
+        inputBinding:
+          prefix: --caller
+          position: 60
+      - "null"
+  calling_parameters:
+    label: "Override default parameters for variant calling"
+    type: string?
+    inputBinding:
+      prefix: --calling_params
+      position: 60
+  min_depth:
+    label: "Minimum depth required to call variant"
+    doc: "Bases with depth below this cutoff will be marked as missing"
+    type: int?
+    inputBinding:
+      prefix: --min_depth
+      position: 60
+  allele_frequency:
+    label: "Minimum allele frequency to call variants"
+    type: float?
+    inputBinding:
+      prefix: --af
+      position: 60
+  dr_allele_frequency:
+    label: "Minimum allele frequency to use variants for prediction"
+    type: float?
+    inputBinding:
+      prefix: --reporting_af
+      position: 60
+  threads:
+    label: "Threads to use"
+    type: int?
+    inputBinding:
+      prefix: --threads
+      position: 60
+  no_trim:
+    label: "Don't trim reads using trimmomatic"
+    type: boolean?
+    inputBinding:
+      prefix: --no_trim
+      position: 60
+  no_flagstat:
+    label: "Don't collect flagstats"
+    type: boolean?
+    inputBinding:
+      prefix: --no_flagstat
+      position: 60
+  no_delly:
+    label: "Don't run delly"
+    type: boolean?
+    inputBinding:
+      prefix: --no_delly
+      position: 60
+  
+  coverage_fraction_threshold:
+    label: "Cutoff used to calculate fraction of region covered by <= this value"
+    type: float?
+    inputBinding:
+      prefix: --coverage_fraction_threshold
+      position: 60
+
+outputs:
+  json_output_file:
+    type: File
+    outputBinding:
+      glob: "$('results/' + inputs.output_prefix + '*.json')"
```

Comment:
```suggestion
      glob: results/$(inputs.output_prefix)*.json
```

Likewise for the rest, and then you don't need the InlineJavascriptRequirement

Reply:
Thanks for the tip!